### PR TITLE
Use `engine_exchangeCapabilities` to log warnings

### LINKED ISSUE #7327: Use `engine_exchangeCapabilities` to log warnings
We no longer call engine methods based on the  `engine_exchangeCapabilities` response after https://github.com/Consensys/teku/pull/7293 . However, we can still use the response from this method to make sure EL supports the methods we expect to be available.

For example when DENEB_FORK_EPOCH is configured for mainnet in CL, we would like ideally EL to return `engine_newPayloadV3` in the exchangeCapabilities response. We could log warnings if not so. This could only be done once on startup to avoid constant polling.

### PR DESCRIPTION
**Pull request by StefanBratanov**
## PR Description

- Enhanced EngineJsonRpcMethodsResolver to return capabilities for all milestones. If a method has been deprecated (added default method in interface) or deleted of course from the code, It will not be returned.
- Create EngineCapabilitiesMonitor which does an exchange capabilities check every 10 epochs (could be too much) against EL.
- It checks engine capabilites against the local capabilities. If some are not supported by EL, a warning will be logged.
- Introduce new parameter `--exchange-capabilities-monitoring-enabled`

## Fixed Issue(s)
fixes #7327 

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: ethereum/executionlayer/src/main/java/tech/pegasys/teku/ethereum/executionlayer/EngineCapabilitiesMonitor.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  }
+
+  private SafeFuture<Void> monitor() {
+    final List<String> capabilities = capabilitiesSupplier.get();
+    return executionEngineClient
+        .exchangeCapabilities(capabilities)
+        .thenApply(ResponseUnwrapper::unwrapExecutionClientResponseOrThrow)
+        .thenAccept(
+            engineCapabilities -> {
+              LOG.debug("Engine API capabilities: " + engineCapabilities);
```
Comment (lucassaldanha) [inline ethereum/executionlayer/src/main/java/tech/pegasys/teku/ethereum/executionlayer/EngineCapabilitiesMonitor.java]: Might be worth specifying that these are the method returned from the remote peer. Something like:

```
LOG.debug("Engine API capabilities response: " + engineCapabilities);
```

File: ethereum/executionlayer/src/main/java/tech/pegasys/teku/ethereum/executionlayer/EngineCapabilitiesMonitor.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+import org.apache.logging.log4j.Logger;
+import tech.pegasys.teku.ethereum.events.SlotEventsChannel;
+import tech.pegasys.teku.ethereum.executionclient.ExecutionEngineClient;
+import tech.pegasys.teku.ethereum.executionclient.response.ResponseUnwrapper;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.infrastructure.logging.EventLogger;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.Spec;
+
+public class EngineCapabilitiesMonitor implements SlotEventsChannel {
```
Comment (lucassaldanha) [inline ethereum/executionlayer/src/main/java/tech/pegasys/teku/ethereum/executionlayer/EngineCapabilitiesMonitor.java:32]: It looks like we don't have a feature toggle for this. Has this been discussed?

**Review by lucassaldanha — APPROVED** [review]
LGTM. Just a nit comment and a question.

File: teku/src/main/java/tech/pegasys/teku/cli/options/ExecutionLayerOptions.java [hunk 3]
```diff
@@ -127,6 +128,18 @@ public class ExecutionLayerOptions {
...
 
+  @Option(
+      names = {"--exchange-capabilities-monitoring-enabled"},
+      paramLabel = "<BOOLEAN>",
+      description =
+          "Enables querying EL periodically for the Engine API methods it supports. If incompatibility is detected, there will be a warning raised in the logs.",
+      arity = "1",
+      showDefaultValue = Visibility.ALWAYS,
+      fallbackValue = "true",
+      hidden = true)
```
Comment (zilm13) [inline teku/src/main/java/tech/pegasys/teku/cli/options/ExecutionLayerOptions.java]: if it's not X, we shouldn't hide it?

Comment (StefanBratanov) [reply teku/src/main/java/tech/pegasys/teku/cli/options/ExecutionLayerOptions.java]: good catch. I just copied above carelessly.

Comment (StefanBratanov) [reply teku/src/main/java/tech/pegasys/teku/cli/options/ExecutionLayerOptions.java]: fixed in last commit + fixed arity as well in this and other classes. There were couple of wrong ones.

File: ethereum/executionlayer/src/main/java/tech/pegasys/teku/ethereum/executionlayer/MilestoneBasedEngineJsonRpcMethodsResolver.java [hunk 4]
```diff
@@ -119,4 +122,13 @@ public <T> EngineJsonRpcMethod<T> getMethod(
     }
     return foundMethod;
   }
+
+  @Override
+  public List<String> getCapabilities() {
```
Comment (tbenr) [inline ethereum/executionlayer/src/main/java/tech/pegasys/teku/ethereum/executionlayer/MilestoneBasedEngineJsonRpcMethodsResolver.java]: I wonder if in theory we should return `Set` to eliminate potential duplication (if a method is used by multiple milestones)

**Review by tbenr — APPROVED** [review]
LGTM. just one comment

File: ethereum/executionlayer/src/main/java/tech/pegasys/teku/ethereum/executionlayer/MilestoneBasedEngineJsonRpcMethodsResolver.java (hunk 4 above)
Comment (StefanBratanov) [reply ethereum/executionlayer/src/main/java/tech/pegasys/teku/ethereum/executionlayer/MilestoneBasedEngineJsonRpcMethodsResolver.java]: Yes. Makes sense. Will change.

**Comment by bgravenorst** [comment]
Removed the doc-change-required flag because the option is documented here: https://docs.teku.consensys.net/reference/cli#exchange-capabilities-monitoring-enabled

**Pull request merged**